src/combinatorics/simple_graph/basic.lean

```diff
@@ -332,4 +332,37 @@ finset.max' (univ.image (λ (v : V), G.degree v)) (nonempty.image univ_nonempty
 
 end finite
 
+section complement
+
+/-!
+## Complement of a simple graph
+
+This section contains definitions and lemmas concerning the complement of a simple graph.
+-/
+
+/--
+We define `complement G` to be the `simple_graph V` such that no two adjacent vertices in `G`
+are adjacent in the complement, and every nonadjacent pair of vertices is adjacent
+(still ensuring that vertices are not adjacent to themselves.)
+-/
+def complement (G : simple_graph V) : simple_graph V :=
+{ adj := λ v w, v ≠ w ∧ ¬G.adj v w,
+  sym := λ v w h,
+    begin
+      cases h with hne hnadj,
+      split,
+      symmetry,
+      exact hne,
+      by_contra,
+      apply hnadj (G.sym h),
+    end,
+  loopless := λ v,
+    begin
+      by_contra,
+      cases h with hne hnadj,
+      apply hne rfl,
+    end }
```

Comment:
I think the mathlib style is to have the `begin` on the same line as the lambda expressions, and then `end` at the same indentation as `sym`.

Here's a golfed version: :golfing: 
```lean
def complement (G : simple_graph V) : simple_graph V :=
{ adj := λ v w, v ≠ w ∧ ¬G.adj v w,
  sym := λ v w ⟨hne, _⟩, ⟨hne.symm, by rwa edge_symm⟩,
  loopless := λ v ⟨hne, _⟩, false.elim (hne rfl) }
```

Reply:
thanks!